apps/instruction_model.py

```diff
@@ -0,0 +1,86 @@
+from state_encoder import StateEncoder
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+class InstructionModel(nn.Module):
+
+    def __init__(self,
+                device,
+                embedding_dim,
+                encoder_dim=128,
+                grid_onehot_size=7,
+                dropout=0.5):
+        super(InstructionModel, self).__init__()
+
+        self.device = device
+
+        #TO DO: xingnan's StateEncoder has different dim with lei's StateEncoderA 
```

Comment:
My StateEncoder feeds into decoder directly, so there probably no need of the last FC layer, whereas IL training requires different input dimension in the original code.

In the original code SimpleNetwork (IL) and LanguageWithAttentionSUM (LSTM) has different encoder layer (e.g. some layer has ReLu in between, some don't). I think the original implementation is wrong and we should try to combine them.

Reply:
With your StateEncoder, I got output shape (64, 27, 9), I tried to sum it to (64, 9) but didn't get it to work. I can tell how you get it, not sure which is the right way.